### PR TITLE
adding defaultChecked property and changing checked property to take a boolean

### PR DESCRIPTION
**Pull request by foopq**
Taking an opinionated stand on https://github.com/purescript-contrib/purescript-react/issues/81 (other bindings have checked as a boolean, such as https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/react/react.d.ts).

Also adding defaultChecked as per https://facebook.github.io/react/docs/forms.html#default-value

### CODE REVIEW

**Comment by ethul** [comment]
I am okay with using `Boolean` in place of `String` for this prop. However, I wonder if we should do this across the board for any prop that accepts a `Boolean`.

@paf31 What are your thoughts on this?

**Comment by paf31** [comment]
Yes, agreed, it'd be better to do this everywhere, although it will be a breaking change.

**Comment by ethul** [comment]
True, it would be breaking. But it would fine-tune the props. I think it could be a good way to go.

@foopq Would you possibly be willing to make this part of your PR?

**Comment by paf31** [comment]
> fine-tune the props

As in reduce the space of valid values? Yes, I think we probably should do it. As for a release, maybe we can wait a little since we need to make a breaking release for 0.10 soon anyway.

Thanks for merging these.

**Comment by ethul** [comment]
Yep, reduce the space of valid values. Sounds good on the release. And no
problem on merging.

**Comment by foopq** [comment]
I'm happy including more in this PR. I'll try to go through the properties tomorrow night and change the others that should be boolean.

**Comment by ethul** [comment]
Wonderful, thank you!

**Comment by foopq** [comment]
Using the signatures from DefinatelyTyped and the React supported attributes page, I've updated the properties here.

**Comment by ethul** [comment]
Looks great! Thank you very much for doing all of this. I will take a closer look and merge this in today or tomorrow.

File: src/React/DOM/Props.purs [hunk 1]
```diff
@@ -168,9 +177,27 @@ icon = unsafeMkProps "icon"
 _id :: String -> Props
 _id = unsafeMkProps "id"
 
+inputMode :: String -> Props
+inputMode = unsafeMkProps "inputMode"
+
+integrity :: String -> Props
+integrity = unsafeMkProps "integrity"
+
+is :: String -> Props
```
Comment (ethul) [inline src/React/DOM/Props.purs]: I don't believe that this is an HTML element. From what I have seen it pertains to polymer. Should this be included? Can someone be using polymer with React? If so, perhaps such attributes should be in a separate module?

File: src/React/DOM/Props.purs [hunk 2]
```diff
@@ -99,13 +99,22 @@ coords = unsafeMkProps "coords"
 crossOrigin :: String -> Props
 crossOrigin = unsafeMkProps "crossOrigin"
 
+dataAttr :: String -> Props
```
Comment (foopq) [inline src/React/DOM/Props.purs]: For these attributes you're asking about, I'm going by what's listed on https://facebook.github.io/react/docs/tags-and-attributes.html. Some of them I'm unfamiliar with as well.

**Comment by ethul** [comment]
Looks great! Thanks so much for this addition.

**Comment by foopq** [comment]
My pleasure :)

**Pull request merged**